leaderboardsgg/leaderboard-site
# feat/LBGG-565: Password recovery flow

## What
This enables a crucial piece of the platform, being able to recover an account and successfully change your password.


## Link to Issue

Closes https://github.com/leaderboardsgg/leaderboard-site/issues/565

## Acceptance

### Steps for testing

- Sign up for an account
- Confirm the associated email
- Logout
- Click the login button
- Click forgot password
- [ ] If you fill out the fields with invalid values, the button should stay disabled
- [ ] Filling out the fields correctly enables the reset button
- Click the reset button
- Click the reset link in the email
- [ ] If the reset token is invalid, you should be navigated back home
- [ ] If it's valid, you should be let onto the reset page
- [ ] If you fill out the fields with invalid values, the change password button should stay disabled
- [ ] Filling out the fields correctly enables the password button
- Click the change password button
- [ ] If the password is invalid, you should get an error
- [ ] If the password is valid, you should get a success modal, and be redirected home

### Images

| Desktop | Tablet | Mobile |
| --- | --- | --- |
| ![image](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/162a6645-4c79-40b9-9dfc-6f65bad2da52) | ![Screen Shot 2024-02-10 at 00 17 27](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/cda6627d-54dd-4472-87e5-54438e7c0c47) | ![Screen Shot 2024-02-10 at 00 18 32](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/fa030237-f4b1-4c7e-a1ae-9b571bbb5a41) |
| ![image](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/20f4ce8c-aacb-4261-afe9-d18daa0205d1) | ![Screen Shot 2024-02-10 at 00 20 36](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/ce0a98f1-a598-46fc-ba3a-d38e75f70ac1) | ![Screen Shot 2024-02-10 at 00 20 29](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/74aa8b91-ec5c-4dd0-8e9e-013158f3e0d4) |
| ![image](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/7bd094c8-89f6-4f61-b7d1-a53e6e3dd091) | ![Screen Shot 2024-02-10 at 00 16 56](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/402911d0-496a-4853-8214-a7540e584fa1) | ![Screen Shot 2024-02-10 at 00 14 39](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/6b93a1d6-4a36-4e3c-addc-b33d41a537d2) |
| ![image](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/f3170ece-bb5d-4097-923c-6a4ecce4c500) | ![Screen Shot 2024-02-10 at 00 29 47](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/fff0a6f1-ce42-4cb8-847f-337707fabe46) | ![Screen Shot 2024-02-10 at 00 29 53](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/5a58a761-576f-4e18-918b-0811e3053e04) |
| ![image](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/18008ed7-fca4-41bd-bb0e-43f204f7d3e6) | ![Screen Shot 2024-02-10 at 00 22 26](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/1489c2ce-7035-4ea2-8e25-537b2e04ea9c) | ![Screen Shot 2024-02-10 at 00 22 35](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/955ff287-cdb2-418f-b6a9-99c032249924) |
| ![image](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/c1b9d258-9d48-4c1d-8682-9e3725f7e362) | ![Screen Shot 2024-02-10 at 00 25 11](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/dcb49d04-bb4f-4848-b028-99b2daac8fda) | ![Screen Shot 2024-02-10 at 00 25 04](https://github.com/leaderboardsgg/leaderboard-site/assets/17736056/2fe8147c-c418-493d-94d9-e38b115127ef) |
